src/osd/osd_types.cc

```diff
@@ -4404,6 +4404,9 @@ void PushOp::encode(bufferlist &bl) const
   ENCODE_START(1, 1, bl);
   ::encode(soid, bl);
   ::encode(version, bl);
+  ::encode(compression, bl);
+  ::encode(src_len, bl);
+  ::encode(src_data_crc, bl);
```

Comment:
new fields need to go at the end, and we need to bump the version in the ENCODE_START macros.  but, i'm not sure we want to add this here...